### PR TITLE
[RNMobile] Extract duplicate e2e test setup to jest config

### PR DESCRIPTION
**Pull request by ceyhun**
## Description

Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2846

- Extracted all the duplicate code between e2e tests into jest config files like `setupFilesAfterEnv` and `testEnvironment`.
- Removed `platformVersion` version from iOS local tests to pick up any available simulator version without hardcoding.

## How has this been tested?

- CI: React Native E2E Tests should pass
- Locally: `npm run native test:e2e:ios:local` and `npm run native test:e2e:android:local` should work as before.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by cameronvoell** [comment]
I merged in latest from master to bring in this CI fix from gutenberg master => https://github.com/WordPress/gutenberg/pull/27347

**Comment by ceyhun** [comment]
Thanks @cameronvoell! They failed again so I restarted and now they seem to pass.

**Comment by ceyhun** [comment]
@cameronvoell got another merge from master and updated the newly added file block tests. Can you please approve if it looks good.

File: packages/react-native-editor/__device-tests__/helpers/utils.js [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
+const wd = require( 'wd' );
+const crypto = require( 'crypto' );
+const path = require( 'path' );
 /**
  * Internal dependencies
  */
-import serverConfigs from './serverConfigs';
-import { iosServer, iosLocal, android } from './caps';
-import AppiumLocal from './appium-local';
+const serverConfigs = require( './serverConfigs' );
```
Comment (cameronvoell) [inline packages/react-native-editor/__device-tests__/helpers/utils.js:12]: Curious why you converted import to require here and in `packages/react-native-editor/__device-tests__/pages/editor-page.js `. Feels like maybe theres a reason I'm missing.

**Review by cameronvoell — APPROVED** [review]
Tests passing locally, and changes look good. I left one question, but otherwise this looks like a much neater setup, and removing the iOS platformVersion requirement is helpful as well. 👍

File: packages/react-native-editor/__device-tests__/helpers/utils.js (hunk 1 above)
Comment (ceyhun) [reply packages/react-native-editor/__device-tests__/helpers/utils.js:12]: That's a good question! These files are now running in a different jest environment than before, and it's currently not possible to apply the same babel transforms/presets we do to the tests themselves. A PR merged to jest resolves this issue and will hopefully be released in the next versions. https://github.com/facebook/jest/pull/8751

Comment (cameronvoell) [reply packages/react-native-editor/__device-tests__/helpers/utils.js:12]: got it, thanks!

**Pull request merged**